src/public/media.ts

```diff
@@ -390,12 +424,30 @@ export namespace media {
     if (!callback) {
       throw new Error('[select Media] Callback cannot be null');
     }
+
     ensureInitialized(FrameContexts.content, FrameContexts.task);
     if (!isAPISupportedByPlatform(mediaAPISupportVersion)) {
       const oldPlatformError: SdkError = { errorCode: ErrorCode.OLD_PLATFORM };
       callback(oldPlatformError, null);
       return;
     }
+
+    if (isMediaCallForVideoAndImageInputs(mediaInputs)) {
+      if (
+        GlobalVars.hostClientType === HostClientType.desktop ||
+        GlobalVars.hostClientType === HostClientType.web ||
+        GlobalVars.hostClientType === HostClientType.rigel
```

Comment:
This approach for determining API availability is really messy and fragile. @sachingupta, can you please take a look at this when you get a chance and advise on the best path forward? I am really worried that we are sprinkling a bunch of this kind of code throughout our SDK and this is creating an extremely fragmented platform. I don't want us to add any more functionality here until we can sort this out.

Reply:
Resolving this conversation as discussed on the Teams **Client SDK API Review** channel post.